wp-includes/class-wp-service-workers.php

```diff
@@ -123,6 +371,27 @@ public function serve_request( $scope ) {
 		echo $this->output; // phpcs:ignore WordPress.XSS.EscapeOutput, WordPress.Security.EscapeOutput
 	}
 
+	/**
+	 * Add logic for routes caching to the request output.
+	 *
+	 * @todo Handle conflicts between routes.
```

Comment:
@westonruter Any thoughts on what would be the goal with detecting conflicts? Should we just report it in the console, for example as Ryan suggested here: https://github.com/xwp/pwa-wp/issues/5#issuecomment-401248790? Or should we report it upon the plugin activation that causes conflicts? Or should we try to handle these in some way, for example if conflicting routes have the same strategy then that's actually not a conflict, however, if there is a conflict, should we dismiss both routes? Or should we leave detecting conflicts out of the prototype for now?

Thoughts?

Reply:
@miina I don't think we can do it in PHP, since we won't be able to programmatically know if two patterns will match without the URL they are matching against. We need to do it at runtime instead via JS. In particular, we can create a `CustomRouter` that overrides `_findHandlerAndParams()` as Jeff suggests in https://github.com/xwp/pwa-wp/issues/5#issuecomment-403082892

Reply:
@westonruter Saw Jeff's comment and was looking into doing it via JS with the `CustomRouter`, I should have asked the question in general comments thread for clarity, sorry for that. Saw how Workbox works by default and takes the first matching route. Considering creating a custom router that overrides `_findHandlerAndParams()` wondering what would be the goal with detecting conflicts and if we should include this into the prototype.
For example should we:
- just leave the default behavior, report the conflicts;
- Dismiss any conflicting logic, report the conflicts;

In case of including it to the prototype, perhaps we could start with leaving the default behavior and reporting conflicts? Thoughts?

Reply:
Note: added custom router initial implementation within fc751ad.